debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsSnapshotProducerIT.java

```diff
@@ -480,6 +487,118 @@ public void shouldNotSnapshotMaterializedViews() throws Exception {
+                new SchemaAndValueField("area", SchemaBuilder.float64().optional()
+                        .parameter(TestHelper.TYPE_NAME_PARAMETER_KEY, "FLOAT83")
+                        .parameter(TestHelper.TYPE_LENGTH_PARAMETER_KEY, "8")
+                        .parameter(TestHelper.TYPE_SCALE_PARAMETER_KEY, "3")
+                        .build(), 12345.123));
+
+        consumer.process(record -> assertReadRecord(record, Collect.hashMapOf("public.alias_table", expected)));
+    }
+
+    @Test
+    @FixFor("DBZ-1413")
+    public void shouldSnapshotDomainAliasWithProperModifiers() throws Exception {
+        TestHelper.dropAllSchemas();
+        TestHelper.execute("CREATE DOMAIN varbit2 AS varbit(3);");
+        TestHelper.execute("CREATE TABLE alias_table (pk SERIAL, value varbit2 NOT NULL, PRIMARY KEY(pk));");
+        TestHelper.execute("INSERT INTO alias_table (value) values (B'101');");
+
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, DecimalHandlingMode.DOUBLE)
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        List<SchemaAndValueField> expected = Collections.singletonList(
+                new SchemaAndValueField("value", Bits.builder(3).build(), new byte[]{ 5, 0 }));
+
+        consumer.process(record -> assertReadRecord(record, Collect.hashMapOf("public.alias_table", expected)));
+    }
+
+    @Test(expected = ComparisonFailure.class)
+    @FixFor("DBZ-1413")
+    public void shouldNotSnapshotNestedDomainAliasTypeModifiersNotPropagated() throws Exception {
+        // The pgjdbc driver does not currently provide support for type modifier resolution
+        // when a domain type extends an existing domain type that extends a base type using
+        // explicit type modifiers.
+        TestHelper.execute("CREATE DOMAIN varbit2 AS varbit(3);");
+        TestHelper.execute("CREATE DOMAIN varbit2b AS varbit2;");
+        TestHelper.execute("CREATE TABLE alias_table (pk SERIAL, value varbit2b NOT NULL, PRIMARY KEY (pk));");
+        TestHelper.execute("INSERT INTO alias_table (value) values (B'101');");
+
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, DecimalHandlingMode.DOUBLE)
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        // We would normally expect that the value column was created with a length of 3.
+        // However due to how the resolution works in the driver, it returns 2147483647.
+        // We probably want to avoid supporting this behavior for now?
```

Comment:
I'm a bit confused by the question. We cannot support nested domain types atm. if that's what it's asking? We also might just remove this test, this limitation is described in the docs.

Reply:
That is actually an old comment/question that I'll remove since the driver fix makes this moot.